source/FreeRTOS_IPv6_Utils.c

```diff
@@ -285,6 +285,141 @@ size_t usGetExtensionHeaderLength( const uint8_t * pucEthernetBuffer,
 }
 /*-----------------------------------------------------------*/
 
+/**
+ * @brief Every IPv6 end-point has a solicited node multicast address and a corresponding
+ * multicast MAC address. The IPv6 solicited node address also has an MLD reports associated
+ * with it. This function manages both the MAC address and the MLD report associated with
+ * the end-point's solicited-node address. On network UP, this function registers the MAC address
+ * with the network driver's filter and creates and MLD report for the UPv6 multicast address.
+ * On network DOWN, the function unregisters the MAC address and removes the MLD report.
+ * This is a "convenience" function that keeps all these tasks under one roof for easier maintenance.
+ *
+ * @param[in] pxEndPoint The end-point for which a network up/down event is being handled.
+ * @param[in] xNetworkGoingUp pdTRUE when the network goes UP, pdFALSE when the network goes DOWN.
+ */
+void vManageSolicitedNodeAddress( struct xNetworkEndPoint * pxEndPoint,
+                                  BaseType_t xNetworkGoingUp )
+{
+    configASSERT( pxEndPoint != NULL );
+    configASSERT( pxEndPoint->pxNetworkInterface != NULL );
+
+    /* do{}while(0) to allow for the use of break statements */
+    do
+    {
+        IPv6_Type_t xAddressType = xIPv6_GetIPType( &( pxEndPoint->ipv6_settings.xIPAddress ) );
+        MACAddress_t xMACAddress = { {
+                                         0x33U,
+                                         0x33U,
+                                         0xFFU,
+                                         pxEndPoint->ipv6_settings.xIPAddress.ucBytes[ 13 ],
```

Comment:
Initializer lists must be constant compile time expressions.

Reply:
I rearranged things a bit to address your concern and also separate declarations and code.